python-sdk/src/astro/files/locations/__init__.py

```diff
@@ -19,7 +19,7 @@ def create_file_location(path: str, conn_id: Optional[str] = None) -> BaseFileLo
     :param path: Path to a file in the filesystem/Object stores
     :param conn_id: Airflow connection ID
     """
-    filetype: FileLocation = BaseFileLocation.get_location_type(path)
+    filetype: FileLocation = BaseFileLocation.get_location_type(path, conn_id)
```

Comment:
I don't think we should enforce getting location type from conn_id. Filepath is fine enough.

How we authorize to that filetype is an implementation detail

Reply:
We are not using `conn_id` to determine the file location type. When the user explicitly passes `conn_id`  the intent here is that the file path and connection passed are aligned.

Reply:
Changed the approach as discussed.